docs/release/controllers.md

```diff
@@ -0,0 +1,178 @@
+[Go module version numbering](https://go.dev/doc/modules/version-numbers) conventions:
+
+- `vX.Y.Z-RC.W` release candidates e.g. `v1.0.0-RC.1`
+- `vX.Y.Z` stable releases e.g. `v1.0.0`
+
+The API versioning and controller versioning are indirectly related. For example,
+a source-controller minor release `v1.1.0` can introduce a new API version
+`v1beta1` for a Kind `XRepository` in the `source.toolkit.fluxcd.io` group.
+
+### Release candidates
+
+Release candidates are intended for testing new features or improvements before a final release.
+
+In most cases, a maintainer will publish a release candidate of a controller for Flux users
+to tests it on their staging clusters. Release candidates are not meant to be deployed in production
+unless advised to do so by a maintainer.
+
+### Patch releases
+
+Patch releases are intended for critical bug fixes to the latest minor version, such as addressing security
+vulnerabilities or fixes to severe problems with no workaround.
+
+Patch releases do not contain breaking changes, feature additions or any type of user-facing changes.
+If a CVE fix requires a breaking change, then a minor release will provide the fix.
+
+We expect users to be running the latest patch release of a given minor release as soon as the
+controller release is included in a Flux patch release.
+
+### Minor releases
+
+Minor releases are intended for backwards compatible feature additions and improvements.
+Note that breaking changes may occur if required by a security vulnerability fix.
+
+Minor releases are used when updating Kubernetes dependencies such as `k8s.io/api` from one minor version to another.
+
+In effect, this means a minor version will be released for all Flux controllers approximately every four months
+after each Kubernetes minor version release. To properly validate the controllers against the latest Kubernetes version,
+we reserve a time window of at least two weeks for Flux controllers end-to-end testing. 
+
+### Major releases
+
+Major releases are intended for drastic changes in the controller behaviour or security stance.
+
+A controller major release will be announced ahead of time throughout all communication channels,
+and a support window of one year will be provided for the previous major version.
+
+## Release cadence
+
+Flux controllers follow Kubernetes three releases per year cadence. After each Kubernetes minor release,
+all controllers are tested against the latest Kubernetes version and are released at approximately two
+weeks after Kubernetes. The newly released controllers offer support for Kubernetes N-2 minor versions.
+
+A Flux controller may have more than three minor releases per year, if maintainers decide to ship a 
+new feature or optimisation ahead of schedule.
+
+## Supported releases
+
+For Flux controllers we support the last three minor releases.
+
+Security fixes, may be backported to those three minor versions as patch releases,
+depending on severity and feasibility.
```

Comment:
Security patching and back-porting to the previous 2 minor releases, across all controllers, may be quite a big undertake given the size of the security team when considering all the additional work required around CVE handling and CVD.

From an OSS perspective, I think back porting to a single minor release sounds more achievable. Companies providing enterprise support for Flux users could bridge that gap for users requiring longer back-porting support.

Reply:
> Security patching and back-porting to the previous 2 minor releases, across all controllers, may be quite a big undertake given the size of the security team 

I don't think the actual back porting is a task that only the security team members can do. A security team member can create an issue asking the maintainers of a controller to do the back port and release. Now it's possible that maintainers don't have spare time to perform this task and no other contributor volunteers to do the back port. If this is the case, then back porting is not feasible due to a shortage of contributors, which is covered by the "depending on severity and feasibility".  I could expend on this if you want and explicitly call out "best-effort / community-driven".

Reply:
My main concern is about the security assurances we provide to our users as part of our CVE handling process.
During the embargo window the security team keep the CVE information private and patch all the supported versions - which currently is only one. It is only after the embargo is lifted and the CVE is public, that it could be back-ported on a "best-effort / community-driven" way. But by that time, users using older versions would already be exposed to an unpatched public CVE. 

Now for both security and bug fixes, the more we expand on this promise, the lesser our long-term capacity. 
I think we should pledge a more sustainable version range, and then mention that back-porting to older versions would be:
- provided by the community in a best-effort basis.
- provided by companies focused on Flux products and Services directed at enterprises (as in [this](https://fluxcd.io/support/#my-employer-needs-additional-help)).

But it would be good to hear from other security team members (cc @hiddeco @scottrigby).

Reply:
Think there is a separation here between two things:

1. CVEs which are directly the result of (not so) happy mistakes in our code.
2. CVEs from upstream dependencies.

---

For scenario 1, having means to automatically back-port things through GitHub labeling may work in about 75% of the cases. I think we could pull this of in a relatively effortless way, as I have seen other projects do.

In terms of this exposing indicators around security issues, the same could be said for closely following commits and/or pull requests. In addition, given it is likely that once this is in place we also leverage this to fix critical (non-security) bugs. Any indicators would blend together with non-CVE things.

This makes me lean towards the idea of this being a non-issue in terms of exposure, or at least outweighing the discomfort of people having to closely follow our minor ranges. Especially given our (at times) aggressive release cadence.

---

For scenario 2, I think this is more of an issue in terms of human workload. Every minor range of a controller may have their own supported minor range of some dependencies, and some of these dependencies may or may not have their procedures around back-porting security fixes. This means that:

1. We have to manually update versions within the minor range of the dependency.
2.  We may not be able to resolve the issue, as it would require updating beyond the point of a version range of the dependency which is accepted within the range of the controller.
3. All of this would be an exponential effort.

---

All together, I think that we should pledge `n-2` for scenario 1, but keep scenario two at `n-1` + best effort.